src/components/Stepper.tsx

```diff
@@ -15,20 +15,31 @@ interface StepperBarProps {
   // The 'value' of the Step that should be displayed
   currentStep: Step["value"];
   onChange: (stepValue: string) => void;
+  resetProgress?: boolean;
 }
 
-export function Stepper({ steps, currentStep, onChange }: StepperBarProps) {
+export function Stepper({ steps, currentStep, onChange, resetProgress }: StepperBarProps) {
   if (steps.length === 0) {
     throw new Error("Stepper must be initialized with at least one step");
   }
 
-  const [progressBarStep, setProgressBarStep] = useState(0);
+  const [progressBarStep, setProgressBarStep] = useState(-1);
+
   useEffect(() => {
-    const currentStepIndex = steps.findIndex((s: Step) => s.value === currentStep) || 0;
-    if (currentStepIndex > progressBarStep) {
-      setProgressBarStep(currentStepIndex);
+    if (resetProgress) {
+      return setProgressBarStep(-1);
```

Comment:
So how is this going to work? Whoever implements `<Stepper />` is going to flash ⚡  a `resetProgress` prop for a moment to reset the progress?

Reply:
I'm not sold on this imp but the reason is so that when a user navigates to the `lots` route, the stepper can render without a progress bar being "filled" in since the "step" is no longer associated with a project so we don't know what is complete ... and therefore, where the max progress bar "fill" should be.

This prop was added to allow the implementor to "reset" the progress bar for instances where the user is on say "options" (or something past "lots") then navigates to "lots" and we need to "clear" the progress bar.

Quick and maybe a little dirty but seems to work for the time being

Reply:
I agree flashing a reset prop is odd; re-reading the code, I'm wondering if we need `progressBarStep` as a separate `useState` at all, and instead should just infer/calc the `progressBarStep` as "iterate through `steps`, and until we hit an `incomplete` step...

Seems like that would handle both "racheting up" as the steps in the `steps` prop become progressively `state: complete`, as well as racheting back down when they go back to `state: incomplete`.

Reply:
yeah good point - `useState` doesn't seem necessary and I've updated to remove the `reset` prop and just recalc